### PR TITLE
Cloudwatch alarms

### PR DESCRIPTION
**Pull request by rajkumar-moengage**

### CODE REVIEW

File: cloudwatch/sqs_alarms/sqs_alarms.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+resource "aws_cloudwatch_metric_alarm" "reports-sqs-msgs-alarm" {
```
Comment (sudip-moengage) [inline cloudwatch/sqs_alarms/sqs_alarms.tf:1]: name should be generic. please don't include business name here.

File: load_balancers/application/alb.tf [hunk 2]
```diff
@@ -96,3 +96,13 @@ resource "aws_lb_listener" "http" {
   }
 }
 
+module "alb_default_alarm" {
+  count                   = var.alb_default_alarm_creation ? 1 : 0
```
Comment (sudip-moengage) [inline load_balancers/application/alb.tf:100]: change variable name to `create_default_alrms`

File: load_balancers/application/variables.tf [hunk 3]
```diff
@@ -184,3 +184,29 @@ variable "vpc_id" {
   description = "VPC ID"
 }
 
+variable "alb_default_alarm_creation" {
+  default     = 0
+  description = "1 to enable default alarms for the alb. 0 to ignore"
+}
+
+variable "alarm_actions" {
+  type        = list(any)
```
Comment (sudip-moengage) [inline load_balancers/application/variables.tf:193]: Optional variable

File: cloudwatch/sqs_alarms/variables.tf [hunk 4]
```diff
@@ -0,0 +1,31 @@
+variable "alarm_actions" {
+  type        = list(any)
+  description = "The Alarm actions arn in list"
+
```
Comment (sudip-moengage) [inline cloudwatch/sqs_alarms/variables.tf:4]: Unnecessary extra line

File: cloudwatch/sqs_alarms/variables.tf [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+}
+variable "dimensions_queuename" {
```
Comment (sudip-moengage) [inline cloudwatch/sqs_alarms/variables.tf:6]: No new line between two variable

File: cloudwatch/sqs_alarms/variables.tf [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  type        = string
+  description = "the SQS Queue name"
+}
+variable "sqs_msgs_threshold" {
```
Comment (sudip-moengage) [inline cloudwatch/sqs_alarms/variables.tf:10]: No new line between two variable

File: cloudwatch/sqs_alarms/variables.tf [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  description = "The number of periods over which data is compared to the specified threshold"
+}
+
+variable "datapoints_to_alarm" {
+  type        = number
+  default     = 2
+  description = "The number of datapoints to check before alarming"
+}
+
+
```
Comment (sudip-moengage) [inline cloudwatch/sqs_alarms/variables.tf:26]: Unnecessary extra line

File: cloudwatch/sqs_alarms/variables.tf [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+}
+variable "dimensions_queuename" {
+  type        = string
+  description = "the SQS Queue name"
+}
+variable "sqs_msgs_threshold" {
+  type        = number
+  description = "The threshold for approximate number of messages for the queue"
+}
+variable "evaluation_periods" {
```
Comment (sudip-moengage) [inline cloudwatch/sqs_alarms/variables.tf:14]: No new line between two variable

File: cloudwatch/sqs_alarms/sqs_alarms.tf (hunk 1 above)
Comment (sudip-moengage) [inline cloudwatch/sqs_alarms/sqs_alarms.tf:1]: Move this folder to cloudwatch/sqs

File: cloudwatch/alb_alarms/README.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
+## Requirements
```
Comment (sudip-moengage) [inline cloudwatch/alb_alarms/README.md:1]: Move this folder to ckoudwatch/application_lb

File: cloudwatch/alb_alarms/variables.tf [hunk 10]
```diff
@@ -0,0 +1,29 @@
+variable "alb_name" {
+  type        = string
+  description = "The readable load balancer name. Specify only the load balancer name, no /app/ prefix or numerical suffix. Example, app3-dashboard"
+}
```
Comment (sudip-moengage) [inline cloudwatch/alb_alarms/variables.tf:4]: space between variables should be consistent

File: cloudwatch/alb_tg_alarms/alb_tg_alarms.tf [hunk 11]
```diff
@@ -0,0 +1,77 @@
+resource "aws_cloudwatch_metric_alarm" "tg-5xx-alarm" {
```
Comment (sudip-moengage) [inline cloudwatch/alb_tg_alarms/alb_tg_alarms.tf:1]: Move this folder to cloudwatch/target_group

File: cloudwatch/alb_tg_alarms/variables.tf [hunk 12]
```diff
@@ -0,0 +1,44 @@
+variable "tg_name" {
```
Comment (sudip-moengage) [inline cloudwatch/alb_tg_alarms/variables.tf:1]: Variable space should be consistent, we follow an extra empty new line between variables

File: cloudwatch/sqs_alarms/README.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
+## Requirements
```
Comment (sudip-moengage) [inline cloudwatch/sqs_alarms/README.md:1]: Move this folder to cloudwatch/sqs

**Review by sudip-moengage — CHANGES_REQUESTED** [review]

File: cloudwatch/sqs_alarms/sqs_alarms.tf (hunk 1 above)
Comment (rajkumar-moengage) [reply cloudwatch/sqs_alarms/sqs_alarms.tf]: why inside "sqs" ?. already its inside "sqs_alarms" right

File: load_balancers/application/alb.tf [hunk 14]
```diff
@@ -96,3 +96,13 @@ resource "aws_lb_listener" "http" {
   }
 }
 
+module "alb_default_alarm" {
+  count                   = var.create_default_alarms ? 1 : 0
+  source                  = "git@github.com:moengage/tf-core.git//cloudwatch/alb_alarms?ref=master"
```
Comment (sudip-moengage) [inline load_balancers/application/alb.tf:101]: This need to point to correct path

File: load_balancers/application/variables.tf (hunk 3 above)
Comment (sudip-moengage) [reply load_balancers/application/variables.tf:193]: It should be optional. Else if create_default_alarms is false, still it will ask for alarm_actions

**Review by sudip-moengage — CHANGES_REQUESTED** [review]

**Review by sudip-moengage — CHANGES_REQUESTED** [review]
The main filename should be main.tf or alarm.tf not sqs_alarm.tf


Resource name should also be _ separated not - separated.

Sorry that I couldn't mention these in earlier reviews

**Review by sudip-moengage — APPROVED** [review]

**Pull request merged**